File: bindings/react-native/ios/HologramViewManager.m
#import <React/RCTViewManager.h>
#import <React/RCTUIManager.h>

// Bridges the Swift view manager + exposes view props to JS (Paper / classic
// architecture). Content is the canonical scene schema, JSON-encoded; `tilt`
// groups the orientation/interaction toggles, likewise JSON. Both are remapped
// onto the `setSceneJson:` / `setTiltJson:` setters added in Swift.
@interface RCT_EXTERN_MODULE(HologramViewManager, RCTViewManager)

RCT_REMAP_VIEW_PROPERTY(scene, sceneJson, NSString)
RCT_REMAP_VIEW_PROPERTY(tilt, tiltJson, NSString)

@end

File: bindings/react-native/android/src/main/java/io/github/alexdonh/hologramism/reactnative/HologramViewManager.kt
package io.github.alexdonh.hologramism.reactnative

import com.facebook.react.bridge.ReadableMap
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp
import io.github.alexdonh.hologramism.HologramView

/**
 * Paper (classic-architecture) view manager exposing the shared native
 * [HologramView] to React Native. Content is one serializable `scene` map (the
 * canonical scene schema); `tilt` groups the orientation/interaction toggles.
 */
class HologramViewManager : SimpleViewManager<HologramView>() {

    override fun getName(): String = "HologramView"

    override fun createViewInstance(context: ThemedReactContext): HologramView =
        HologramView(context)

    @ReactProp(name = "scene")
    fun setScene(view: HologramView, scene: ReadableMap?) {
        scene?.let { view.setScene(it.toHashMap()) }
    }

    @ReactProp(name = "tilt")
    fun setTilt(view: HologramView, tilt: ReadableMap?) {
        tilt?.let { view.setTilt(it.toHashMap()) }
    }
}
